design-documents/graph-ql/versioning.md

```diff
@@ -0,0 +1,28 @@
+## Problem 
+
+GraphQL APIs are constantly evolving and require changes to be made to existing queries, mutations and data types.
+
+Some GraphQL schema changes can be covered using versionless strategy, which [is considered best practice](https://graphql.org/learn/best-practices/#versioning) for GraphQL APIs. Examples of such changes include:
+ - new fields in return types
+ - deprecation and subsequent removal of fields in return types
+ - new optional arguments of the fields, queries and mutations
+ - replacement of existing query or mutation with a new semantically different one. Example is replacing a set of mutations for adding products to cart with a single unified mutation which can add products of any type to cart. Semantics of the operations is different in this case, that is why it is not difficult to come up with a new meaningful name.
```

Comment:
This items should probably mention that the old mutation should be first deprecated (and continue being supported) and then removed (same as in item 2).